api/src/graphql/Post.ts

```diff
@@ -17,29 +58,75 @@ export const PostQuery = extendType({
       type: "Post",
       resolve(parent, args, context) {
         return context.prisma.post.findMany();
-      }
+      },
+    });
+    t.field("post", {
+      type: "Post",
+      args: {
+        id: nonNull(intArg()),
+      },
+      resolve(parent, args, context) {
+        const { id } = args;
+        return context.prisma.post.findUnique({
+          where: { id },
+        });
+      },
     });
-  }
+    t.nonNull.list.field("unmatchedPosts", {
+      type: "Post",
+      resolve(parent, args, context) {
+        return context.prisma.post.findMany({
+          where: { navigatorId: null },
+        });
+      },
+    });
+  },
 });
 
 export const PostMutation = extendType({
-    type: "Mutation",
-    definition(t) {
-        t.nonNull.field("post", {
-            type: "Post",
-            args: {
-                description: nonNull(stringArg()),
-                title: nonNull(stringArg()),
-            },
-            resolve(parent, args, context) { 
-                const newPost = context.prisma.post.create({
-                    data: {
-                        description: args.description,
-                        title: args.title,
-                    },
-                });
-                return newPost;
+  type: "Mutation",
+  definition(t) {
+    t.nonNull.field("post", {
+      type: "Post",
+      args: {
+        description: nonNull(stringArg()),
+        title: nonNull(stringArg()),
+        requiredSkillsId: nonNull(list(nonNull(intArg()))),
+      },
+      async resolve(parent, args, context) {
+        const { description, title, requiredSkillsId } = args;
+        const { userId } = context;
+
+        if (!userId) {
+          throw new Error("You have to log in");
+        }
+
+        const user = (await context.prisma.user.findUnique({
+          where: { id: userId },
+        })) as User;
+        if (user.matchingPoint == 0) {
```

Comment:
```suggestion
        if (user.matchingPoint < 1) {
```
可能性としてあるかは不明ですが、matchingPointが0以下の場合に弾いていいと思います！